### DISCUSSION TITLE
Add enemy speed option that doesn't show movement but reviews final positions

### DISCUSSION THREAD

**Discussion by Granfalegion** [ihhub/fheroes2#8112] [Ideas]
Some enemy turns take forever to watch play out.  This is _especially_ true in fheroes maps with liths owing to the decision to allow costless lith travel via the spacebar.  The fading-in/out effects of using a lith are not sped up by enemy speed settings, and because an enemy hero can theoretically travel through such liths **indefinitely** on a given turn, the experience is awful.

Some players resolve this issue by disabling the viewing of enemy movement altogether.  This is certainly one available solution.  But it also makes it more likely that you will miss critical movement such as an army suddenly bearing down on a castle or other strategic location.

I propose that (_so long as liths-via-spacebar travel is allowed_) a new Enemy Speed option be introduced: **Review**.  Under this setting, enemy _movement_ is not shown, but that at the end of an enemy turn, the camera would cycle through each of the visible enemy heroes on their final locations.  This draws attention to where all the enemy heroes are now, but doesn't spend any time on how they got there.  For maps where "how they got there" can feel like it takes minutes, this could strike a very helpful middleground.

**Comment by oleg-derevenetz** [comment]
> This is especially true in fheroes maps with liths owing to the decision to allow costless lith travel via the spacebar. [...], and because an enemy hero can theoretically travel through such liths indefinitely on a given turn, the experience is awful.

Spacebar is allowed only for human-controlled heroes. AI-controlled heroes do not have such a privilege and they cannot travel through Stone Liths or Whirlpools indefinitely, even theoretically.

> The fading-in/out effects of using a lith are not sped up by enemy speed settings

#8114 is intended to improve this.

**Reply by Granfalegion** [reply]
> Spacebar is allowed only for human-controlled heroes. AI-controlled heroes do not have such a privilege and they cannot travel through Stone Liths or Whirlpools indefinitely, even theoretically.

This is currently true, but I'm aware that teaching the AI to use spacebar liths has at least reached the point of discussion.

There are several maps with large numbers of liths that lead to a similar effect.  While it's not strictly indefinite without spacebar tech, it can reach to high numbers when they are tightly clustered.  `Timber!` and `THUNK` both come to mind.
